pkg/analysis/optionalfields/analyzer.go

```diff
@@ -0,0 +1,393 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package optionalfields
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+	kalerrors "sigs.k8s.io/kube-api-linter/pkg/analysis/errors"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/extractjsontags"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/inspector"
+	markershelper "sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/markers"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/utils"
+	"sigs.k8s.io/kube-api-linter/pkg/config"
+	"sigs.k8s.io/kube-api-linter/pkg/markers"
+)
+
+const (
+	name = "optionalfields"
+
+	optionalMarker            = markers.OptionalMarker
+	requiredMarker            = markers.RequiredMarker
+	kubebuilderOptionalMarker = markers.KubebuilderOptionalMarker
+	kubebuilderRequiredMarker = markers.KubebuilderRequiredMarker
+
+	minItemsMarker      = markers.KubebuilderMinItemsMarker
+	minLengthMarker     = markers.KubebuilderMinLengthMarker
+	minPropertiesMarker = markers.KubebuilderMinPropertiesMarker
+
+	minimumMarker = markers.KubebuilderMinimumMarker
+	maximumMarker = markers.KubebuilderMaximumMarker
+
+	enumMarker = markers.KubebuilderEnumMarker
```

Comment:
Any particular reason to make these local constants instead of just using the `markers` package constants?